docs/src/design.md

```diff
@@ -30,18 +30,25 @@ One does not have to carry both a [`Model`](@ref) type, and a varying collection
 
 ```julia
 model = StatsBase.fit(
-    template,
+    template::Template,
```

Comment:
Really we're repeating ourselves here. Maybe we should just reference the docstrings here to avoid these getting out of sync?